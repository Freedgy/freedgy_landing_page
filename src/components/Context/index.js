import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Avatar from '@material-ui/core/Avatar';
import Box from '@material-ui/core/Box';
import CardMedia from '@material-ui/core/CardMedia';

const styles = theme => ({
    context: {
        margin: "5% 0%",
        width: "100%",
        height: "100%"
    },
    content: {
        width: "100%",
        height: "100%",
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
    }
});

function Context({ classes, datas }) {
    return (
        <section id="context" className={classes.context}>
            <Typography variant="h2" align="center">
                {datas.title}
            </Typography>
            <Box className={classes.content}>
                <Typography variant="p" align="center">
                    {datas.description}
                </Typography>
            </Box>
        </section>
    );
}

Context.propTypes = {
    classes: PropTypes.object.isRequired,
    datas: PropTypes.object.isRequired,
};

export default withStyles(styles)(Context);