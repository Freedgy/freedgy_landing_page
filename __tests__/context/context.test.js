import React from "react"
import renderer from "react-test-renderer"

import Context from "../../src/components/Context"

describe("Context", () => {
    it("renders correctly", async () => {
        const datas = require("../../static/pages/pages.en.json");

        const tree = renderer
            .create(<Context datas={datas.content.context} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})