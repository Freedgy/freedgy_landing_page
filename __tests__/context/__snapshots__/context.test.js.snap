// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context renders correctly 1`] = `
<section
  className="Context-context-1"
  id="context"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter"
  >
    Context
  </h2>
  <div
    className="MuiBox-root MuiBox-root-3 Context-content-2"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-alignJustify"
    >
      Epitech is a computer school created in 1999 that trains IT Experts. This training sought by companies is based on an innovative model that emphasizes three qualities that are increasingly required: adaptability, self-progression and a project-oriented workforce. To carry out this program, called Grande Ecole program,the students' course is divided into two parts. A first 2-year program will address the most technical topics in order to build a solid base and varied knowledge. The second course of the school is called the innovation cycle. It lasts 3 years, from the third to the fifth year of the Epitech program. It aims to develop a project, an expertise or a technology, and is structured of three stages: the idea and positioning, prototyping and feedback, and finally industrialization and commercialization. This project is called the Epitech Innovation Project (EIP).
    </p>
    <br />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-alignJustify"
    >
      More than 100 Epitech Innovative Projects are presented each year at the Epitech Experience forum, the final highlight of the fifth year and a real synthesis of Epitechâ€™s program. The students come to defend the success of their project in front of an ecosystem of entrepreneurs, decision-makers and investors. Even more than just a project, it is a successful step into working life thanks to two years of teamwork.
    </p>
  </div>
</section>
`;
